src/suggestion.rs

```diff
@@ -50,14 +69,204 @@ impl fmt::Display for Detector {
     }
 }
 
+// For long lines, literal will be trimmed to display in one terminal line.
+// Misspelled words that are too long shall also be ellipsized.
+pub fn condition_display_content(
```

Comment:
Also quite complex, looks fine but I'm not sure if I understand every bit

Reply:
I aim to add more explanation in another PR. Maybe it will be easier with examples.